c10/core/DispatchKey.cpp

```diff
@@ -126,6 +126,25 @@ const char* toString(DispatchKey t) {
     case DispatchKey::TESTING_ONLY_GenericMode:
       return "TESTING_ONLY_GenericMode";
 
+    // Note [Out-of-tree vmap+grad prototype]
+    // The following keys are used in the implementation of the out-of-tree
+    // composable functions transforms (vmap+grad) prototype that lives at
+    // https://github.com/zou3519/functorch
+    // We plan on eventually upstreaming the prototype into core, at which
+    // point it will have a different design that should use fewer keys.
+    case DispatchKey::Python:
```

Comment:
nit: could this one be more specific? Or at least have a less generic name?
If this is specifically for FX or torch_function, maybe we want to mention it? Or even FuncTorchPython if this is specific to this lib.

Reply:
In the end state, we do want a key whose name is Python: https://github.com/pytorch/pytorch/issues/55095. The idea is that all `__torch_function__` objects should have the ability to go into the dispatcher using this key

Reply:
But that would be part of the "upstreaming the prototype into core, at which point it will have a different design" ?
In particular, it is simple to delete a key that is used just for prototype, it might be harder to re-purpose one without changing the name?
But if you think that this is going to be the final behavior, its ok with me.

Reply:
Worst comes to worst we can rename the key. But I do see the concern. What do you think about prefixing all of the keys with "Functorch" to avoid confusion?

Reply:
I think a FuncTorch prefix is good.

It also has the added benefit when you move to core and you have "proper" keys that match, you can simply alias the FuncTorch key to the corresponding core key and you're done :)

Reply:
Cool, will do!